Meadowlark-js/backends/meadowlark-mongodb-backend/src/repository/Db.ts

```diff
@@ -134,3 +127,46 @@ export const asUpsert = (session: ClientSession): ReplaceOptions => ({ upsert: t
 
 // MongoDB FindOption to return at most 5 documents
 export const limitFive = (session: ClientSession): FindOptions => ({ limit: 5, session });
+
+/**
+ * Lock in-use meadowlark documents, both those being directly updated and those being referenced.
+ */
+export async function writeLockDocuments(
+  concurrencyCollection: Collection<ConcurrencyDocument>,
+  concurrencyDocuments: ConcurrencyDocument[],
+  session: ClientSession,
+): Promise<void> {
+  await concurrencyCollection.insertMany(concurrencyDocuments, { session });
+}
+
+/**
+ * Remove the locks on in-use meadowlark documents, both those being directly updated and those being referenced.
+ */
+export async function removeDocumentLocks(
```

Comment:
I'm wondering if this needs to be a separate function. I think we may be able to deleteMany right after we insertMany...

Reply:
Hum... interesting idea, never thought of it. I'll give that a try and experiment with the tests.